### ISSUE TITLE
Double Authorization header when using WS with NTLM authentication

### ISSUE THREAD

**Issue by martinstuder** [playframework/playframework#4002]
Reference: https://groups.google.com/forum/#!topic/play-framework/DUfH6q-QOwY

When trying to call a .NET web service (hosted on IIS) using WS with NTLM authentication, I get an HTTP 400 - "the request has an invalid header name". Investigating the traffic using Wireshark/Fiddler in more detail, I found that WS (async-http-client?) inserts a second "Authorization" header which seems to be rejected by IIS.

The code:
```
val response = client.url("http://myhost.mydomain.com/myservice")
                     .withFollowRedirects(true)
                     .withAuth("<USER>", "<PWD>", WSAuthScheme.NTLM)
                     .get()
```

What happens:

* Initial client request is triggered
```
GET http://myhost.mydomain.com/myservice HTTP/1.1
Host: myhost.mydomain.com
Authorization: TlRMTVNTUAABAAAANQIIIAoACgAyAAAAEgASACAAAABsAG8AYwBhAGwAaABvAHMAdABOAFQATABNACAA
Connection: keep-alive
Proxy-Connection: keep-alive
Accept: */*
User-Agent: NING/1.0
```
Not sure if the authorization cookie should be prefixed with "NTLM" here ...

* Server responses with HTTP 401 Unauthorized
```
HTTP/1.1 401 Unauthorized
Content-Type: text/html
Server: Microsoft-IIS/7.5
WWW-Authenticate: NTLM
X-Powered-By: ASP.NET
Date: Fri, 27 Feb 2015 15:46:22 GMT
Content-Length: 1293
Proxy-Support: Session-Based-Authentication


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>
<title>401 - Unauthorized: Access is denied due to invalid credentials.</title>
<style type="text/css">
<!--
body{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}
fieldset{padding:0 15px 10px 15px;} 
h1{font-size:2.4em;margin:0;color:#FFF;}
h2{font-size:1.7em;margin:0;color:#CC0000;} 
h3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} 
#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:"trebuchet MS", Verdana, sans-serif;color:#FFF;
background-color:#555555;}
#content{margin:0 0 0 2%;position:relative;}
.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}
-->
</style>
</head>
<body>
<div id="header"><h1>Server Error</h1></div>
<div id="content">
 <div class="content-container"><fieldset>
  <h2>401 - Unauthorized: Access is denied due to invalid credentials.</h2>
  <h3>You do not have permission to view this directory or page using the credentials that you supplied.</h3>
 </fieldset></div>
</div>
</body>
</html>
```

* Client re-initiates request based on "WWW-Authenticate: NTLM" response above
```
GET http://myhost.mydomain.com/myservice HTTP/1.1
Host: myhost.mydomain.com
Authorization: NTLM TlRMTVNTUAABAAAANQIIIAAAAAAyAAAAEgASACAAAABsAG8AYwBhAGwAaABvAHMAdAA=
Authorization: TlRMTVNTUAABAAAANQIIIAoACgAyAAAAEgASACAAAABsAG8AYwBhAGwAaABvAHMAdABOAFQATABNACAA
Connection: keep-alive
Connection: keep-alive
Accept: */*
User-Agent: NING/1.0
```

**Notice the duplication of the "Authorization" header!**

* Server responses with HTTP 400 Bad Request
```
HTTP/1.1 400 Bad Request
Content-Type: text/html; charset=us-ascii
Server: Microsoft-HTTPAPI/2.0
Date: Fri, 27 Feb 2015 15:46:22 GMT
Connection: close
Content-Length: 339


<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Bad Request</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY><h2>Bad Request - Invalid Header</h2>
<hr><p>HTTP Error 400. The request has an invalid header name.</p>
</BODY></HTML>
```

**Comment by wsargent** [comment]
Play WS is an async wrapper around AsyncHttpClient, so most likely the problem is that the interaction is not syncing up some how.

Here's the NTLMTest.java for how NTLM should work:

https://github.com/AsyncHttpClient/async-http-client/blob/b85d5b3505d9f6e80d278fef88876f6546e73079/api/src/test/java/org/asynchttpclient/async/NTLMTest.java

Play doesn't do much besides call auth() and pass it through to the builder:

https://github.com/playframework/playframework/blob/2.3.x/framework/src/play-ws/src/main/scala/play/api/libs/ws/ning/NingWS.scala#L226

@martinstuder Can you run this with your Logback settings ramped up to:

```
 <logger name="com.ning.http.client" level="TRACE" />
  <logger name="play.api.libs.ws" level="TRACE" />
```

and see what the output is?

**Comment by jroper** [comment]
@wsargent I'm already across this - NTLM was completely broken in AHC 1.8:

https://github.com/AsyncHttpClient/async-http-client/issues/730

It should be fixed in 1.9.

**Comment by megazord** [comment]
@jroper, @wsargent 

Play is already using the latest version of AHC (1.9.17). Can we close this issue?
